src/lib.rs

```diff
@@ -46,7 +46,20 @@ pub async fn bootstrap(
     let s3_client = get_s3_client(&config).await;
     let geoip_resolver = get_geoip_resolver(&config, &s3_client).await;
 
-    let provider = BlockchainApiProvider::new(config.project_id.clone());
+    let provider = if let Some(blockchain_api_endpoint) = &config.blockchain_api_endpoint {
+        Some(
+            BlockchainApiProvider::new(
+                config.project_id.clone(),
+                blockchain_api_endpoint
+                    .parse()
+                    .expect("Error parsing blockchain_api_endpoint"),
```

Comment:
Is it ok to unwrap here?

Reply:
Yeah because it's in app startup code. I prefer to use `Result`s in `main()` for stylistic reasons, but it doesn't matter so just throwing in a panic real quick here.